docs/source/deployment/nodes_grouping.md

```diff
@@ -0,0 +1,22 @@
+| Namespaces | - Namespaces allow you to group nodes, ensuring clear dependencies and separation within a pipeline while maintaining a consistent structure. <br> - Similar to pipelines or tags, you can enable selective execution using namespaces. <br> - Using namespaces allows you to visualize your grouped nodes through kedro-viz. <br> ![Switching between different namespaces in Kedro Viz](../meta/images/kedro_viz_switching_namespace.png)| - You cannot run multiple namespaces simultaneously—Kedro only allows executing one namespace at a time. <br> - Defining namespace at Node-level: If you define namespaces at the node level, they behave similarly to tags and do not guarantee execution consistency. This level of safety is only ensured when using namespaces at the pipeline level. <br> - Defining namespace at Pipeline-level: When applying a namespace at the pipeline level, Kedro automatically renames all inputs, outputs, and parameters within that pipeline. While this helps with modularization, it can introduce unintended naming conflicts if the pipeline is connected elsewhere or if parameters are referenced outside the pipeline. | - Namespaces are ideal for organizing pipelines and allow you to run only part of a pipeline. <br> - If your pipeline structure is well-defined, converting it into namespaces unlocks better visualization in Kedro-Viz, making it easier to explore and understand your nodes grouping. <br> - Customizing Deployment Groups: If you want to group nodes differently or add additional groups of nodes, adding namespaces at the node level helps. Similar to tags, but unlike tags (which only filter nodes in Kedro-Viz), namespaces allow separate visual representation of grouped nodes. | - Increased complexity: <br> - If your pipeline has only a few nodes, adding namespaces may introduce unnecessary complexity without significant benefits. <br> - Applying namespaces at the pipeline level can make the structure harder to manage, with all the names changing as explained in the column of "What doesn't work with Kedro". | - Run by kedro run --namespace=<your namespace name> <br> - More information: "TBD Namespaces" |
+
+
+## **Choosing the Right Grouping Method**
+
+If you're unsure which method to use, consider these questions:
+
+- **Do you need ...** → Use **Pipelines**
+- **Do you need ...** → Use **Tags**
+- **Do you need ...** → Use **Namespaces**
+
+By choosing the right method, you can optimize your Kedro project for **scalability, readability, and efficient deployment.** 🚀
```

Comment:
This has too much of a marketing vibe to it. If we make statements like this in the docs the least we can do is explain them.

Reply:
ah yeah , agree I went a bit too much on the "sale" haha. Have removed this section completely now 👍